src/utils/spark.py

```diff
@@ -261,6 +263,12 @@ def read(self) -> None:
                     field.name, df[field.name].cast(DecimalType(10, 0))
                 )
 
+        # Add a timestamp column to the data to track when it was loaded
+        df = df.withColumn(
+            "loaded_at",
+            date_format(current_timestamp(), "yyyy-MM-dd HH:mm:ss.SSS"),
+        )
+
```

Comment:
I somehow dropped the `loaded_at` column in an earlier PR, but we need it for dbt source freshness tests and for parity with sqoop.